src/common/utilities/types.ts

```diff
@@ -16,3 +16,34 @@ export const isTimeArray = (value: unknown): value is TimeObject[] => {
   }
   return false;
 };
+
+export const isIdArray = (value: unknown): value is IdObject[] => {
+  if (Array.isArray(value) && value[0] !== null && value[0] !== undefined) {
+    if ('id' in value[0]) {
+      return true;
+    }
+  }
+  return false;
+};
+
+export const isEmployeeTimesheetArray
+ = (value: unknown): value is EmployeeTimesheet[] => {
+   if (Array.isArray(value) && value[0] !== null && value[0] !== undefined) {
+     if ('hours' in value[0] && 'submitted' in value[0] && 'invoiced' in value[0]
+    && 'paid' in value[0] && 'username' in value[0] && 'email' in value[0]) {
+       return true;
+     }
+   }
+   return false;
+ };
```

Comment:
Nice! Another good example of where the guard clauses would be helpful, though.

Rather than:

```
if (a) {
  if (b) {
    return true;
  }
}
return false;
```

This could be:

```
if (!a) {
  return false;
}

if (!b) {
  return false;
}

return true;
```

Or, since we're just returning booleans, you could do something like:

```
  return a && b;
```

The full function in this case would be:

```
export const isEmployeeTimesheetArray = (value: unknown): value is EmployeeTimesheet[] => (
  Array.isArray(value)
  && value[0] !== null
  && value[0] !== undefined
  && 'hours' in value[0]
  && 'submitted' in value[0]
  && 'invoiced' in value[0]
  && 'paid' in value[0]
  && 'username' in value[0]
  && 'email' in value[0]
);
```

Reply:
Also, is this type check really sufficient? What if `value[1]` is not an employee timesheet? We're only checking the first element. And, is `value[0] !== null && value[0] !== undefined` sufficient? Maybe we should use `isObjectRecord` for this to make sure it's actually an object.